graphql_test.go

```diff
@@ -304,6 +323,45 @@ func TestNilInterface(t *testing.T) {
 	})
 }
 
+func TestErrorWithExtension(t *testing.T) {
+	err := withExtensionError{
+		Code:    "NotFound",
+		Message: "This is not the droid you are looking for",
```

Comment:
The test is named `TestErrorWithExtension`, however the error doesn't have an extension field set. Was this indetntional? Can you, please, add some more test cases? Then I'll be happy to merge this PR.

Reply:
Maybe you need to add cases both with and without the extension field 🤔

Reply:
Well you can just return:
```go
err := errors.QueryError{
    Message: "error message goes here",
    Extensions: map[string]interface{}{"key1": "val1", "key2": "val2"},
}

return nil, err
```

Reply:
I guess we have to return some kind of a custom error that implements this interface (this interface will not be part of the library):
```go
type ResolverError interface {
    error
    Extensions() map[string]interface{}
}

```

Then at the top of `internal/exec/exec.go` define this interface:
```go
type extensionser interface {
	Extensions() map[string]interface{}
}
```

and later in `internal/exec/exec.go:190` the code will look like:
```go
if f.field.HasError && !callOut[1].IsNil() {
    resolverErr := callOut[1].Interface().(error)
    err := errors.Errorf("%s", resolverErr)
    err.Path = path.toSlice()
    err.ResolverError = resolverErr
    if ex, ok := callOut[1].Interface().(extensionser) {
        err.Extensions = ex.Extensions()
    }
    return err
}
```

This way it will be up to the user to define any custom error type (e.g. `AuthenticationError`, `ForbiddenError`, `ValidationError`, `ServerError`, etc.) 
Ideally, this will also need to be documented with another example.